### PR TITLE
[FEATURE] Update Recents

### LINKED ISSUE #2912: Fix "Recent addresses" functionality in the Send flow
**Issue:** Currently in the Send flow, under the list of “recent addresses”, you see all the addresses that users have interacted with and that list is populated from the transaction history. This doesn’t match with user expectation, and could lead to users sending assets to addresses they didn’t intend to. 

**Fix:** In the ‘recent addresses’ list of Send flow, ONLY show the addresses that were previously added & sent to in the “recipient address field”.

**QA paths to test:**

- this should solve the issue where people are seeing Swaps contract address (and other contract addresses after interaction with sites) in the “recent addresses”
- this should allow contract addresses if the user had intentionally sent to a contract addresses via the Send flow (e.g., to another contract account wallet, or even for wrapping ETH)
- “Recent addresses” should ONLY show addresses that users have sent to via the “Send” feature in the past (in the order: most recent “to address” top of the list). 

Related: #2822 & https://github.com/MetaMask/metamask-mobile/pull/2823

### PR DESCRIPTION
**Pull request by rickycodes**
**Description**

render recents based on items stored in redux instead of from `transactions`

**Notes for QA**

- when starting off you should have zero recents (this feature is _not_ backwards compatible).
- recents should start populating (in reverse chronological order) after you do a send.
- there should never be more than three recent addresses.
- functionality should otherwise be the same as before.

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Issue**

Resolves #2912

### CODE REVIEW

File: app/components/Views/Login/index.js [hunk 1]
```diff
@@ -491,7 +491,7 @@ class Login extends PureComponent {
 				<TouchableWithoutFeedback onPress={Keyboard.dismiss}>
 					<View style={styles.areYouSure}>
 						<Text style={[styles.heading, styles.delete]}>
-							{strings('login.type_delete', { delete: DELETE })}
+							{strings('login.type_delete', { [DELETE]: DELETE })}
```
Comment (rickycodes) [inline app/components/Views/Login/index.js:494]: unrelated to this, but I wanted to re-utilise this `const` to ensure the `key` is always the string `'delete'`

File: app/components/Views/SendFlow/AddressElement/index.js [hunk 2]
```diff
@@ -80,8 +80,8 @@ class AddressElement extends PureComponent {
 		const secondaryLabel = name && name[0] !== ' ' && renderShortAddress(address);
 		return (
 			<TouchableOpacity
-				onPress={() => onAccountPress(address)} /* eslint-disable-line */
-				onLongPress={() => onAccountLongPress(address)} /* eslint-disable-line */
+				onPress={() => onAccountPress(address)}
+				onLongPress={() => onAccountLongPress(address)}
```
Comment (rickycodes) [inline app/components/Views/SendFlow/AddressElement/index.js:84]: no idea why these were disabled

**Review by gantunesr — APPROVED** [review]
lgtm

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
Fix states = "In the ‘recent addresses’ list of Send flow, ONLY show the addresses that were previously added & sent to in the “recipient address field”."

I sent SAI to to an address and I am not seeing the same address I added in the recipient address field

seen here = https://recordit.co/bgKck6zJlh

image

image

**Comment by ibrahimtaveras00** [comment]
It does work with normal ETH send = http://recordit.co/VrnXby1E6Q

image

**Review by ibrahimtaveras00 — APPROVED** [review]
Fixes look good, QA Passed 👍🏽

**Pull request merged**